addons/im_livechat/models/discuss_channel.py

```diff
@@ -315,3 +315,12 @@ def _action_unfollow(self, partner=None, guest=None, post_leave_message=True):
             # sudo: discuss.channel - last operator left the conversation, state must be updated.
             channel_sudo.livechat_active = False
             self._bus_send_store(self, "livechat_active")
+
+    def add_members(
```

Comment:
It feels like this would be better as an override of member create/unlink, if a non-store compute field member_type which is given to create, which has an inverse that does the job of creating the history (and the compute itself is working to get the current type based on history).